r/R/record-batch.R

```diff
@@ -147,21 +147,41 @@ RecordBatch <- R6Class("RecordBatch", inherit = ArrowObject,
   )
 )
 
-RecordBatch$create <- function(..., schema = NULL){
+RecordBatch$create <- function(..., schema = NULL) {
   arrays <- list2(...)
+  if (length(arrays) == 1 && inherits(arrays[[1]], c("raw", "Buffer", "InputStream", "Message"))) {
+    return(RecordBatch$from_stream(arrays[[1]], schema))
+  }
+  # Else, list of arrays
   # making sure there are always names
   if (is.null(names(arrays))) {
     names(arrays) <- rep_len("", length(arrays))
   }
   stopifnot(length(arrays) > 0)
+  # TODO: should this also assert that they're all Arrays?
   shared_ptr(RecordBatch, RecordBatch__from_arrays(schema, arrays))
 }
 
+RecordBatch$from_stream <- function(obj, schema) {
```

Comment:
This could well cause confusion. `RecordBatch$from_ipc_message` would be the most unambiguous name

Reply:
I don't understand how Message relates to IPC stream, but IIUC these are all about reading a record batch from and IPC stream format?

Reply:
This function loads a RecordBatch from an encapsulated IPC message. So it's loading a single message component of a stream, assuming you already have the schema. Note that it won't work if there are dictionaries (you need to pass a populated) dictionary memo

Reply:
Have a look at https://www.slideshare.net/wesm/apache-arrow-workshop-at-vldb-2019-boss-session-169065658

Slides 35 onward. Basically what is here is a function that loads just the "RecordBatch" message given a schema plus dictionaries that have already been loaded previously